0035-spsp-push-payments/0035-spsp-push-payments.md

```diff
@@ -0,0 +1,147 @@
+| `asset_info` | Object | _(OPTIONAL)_ Details about the agreement's asset. For invoices, this is the currency the invoice is stated in. |
+| `asset_info.code` | String | Asset code to identify the agreement's currency. Currencies that have [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217) codes should use those. |
+| `asset_info.scale` | Integer | The scale of the amounts `balance.maximum` and `balance.current` are stated in (e.g. an amount of `"1000"` with a scale of `2` translates to `10.00` units of the server's asset/currency) |
+| `receiver_info` | Object | _(OPTIONAL)_ Arbitrary additional information about the receiver. This field has no schema and the receiver may include any fields they choose. The field names listed below are recommended merely for interoperability purposes. |
+| `receiver_info.name` | String | _(OPTIONAL)_ Full name of the individual, company or organization the receiver represents |
+| `receiver_info.image_url` | HTTPS URL | _(OPTIONAL)_ URL where the sender can get a picture representation of the receiver |
+
+**Note:** Currency amounts are denominated as integer strings instead of native JSON numbers to avoid losing precision during JSON parsing. Applications MUST represent these numbers in a data type that has precision equal or greater than an unsigned 64-bit integer.
+
+
+##### Errors
+
+###### receiver Does Not Exist
+
+``` http
+HTTP/1.1 404 Not Found
+Content-Type: application/spsp4+json
+
+{
+  "id": "InvalidReceiverError",
+  "message": "Invalid receiver ID"
+}
+```
+
+## Model of Operation
+
+### Conducting the push payment
+
+We assume that the client knows the server's SPSP endpoint (see [Payment Pointers](../0026-payment-pointers/0026-payment-pointers.md)).
+
+1. The user's SPSP client queries the server's SPSP Endpoint.
+
+2. The SPSP endpoint responds with the server info, including the server's ILP address (`destination_account`) and the shared secret (`shared_secret`) to be used in STREAM. 
+    * The `destination_account` SHOULD be used as the STREAM destinationAccount.
+    * The `shared_secret` SHOULD be decoded from base64 and used as the STREAM sharedSecret.
+  
+    It MAY respond with additional information if it is an invoice server or a pull payment server (see [SPSP Pull Payments](../0036-spsp-pull-payments/0036-spsp-pull-payments.md)). If there is no `balance` field or the amounts within `balance` are negative, the client can push to this endpoint.  
+
+3. The SPSP client establishes a STREAM connection to the server using the server's ILP address and shared secret and constructs a stream to the server on the ILP/STREAM connection.
+
+4. The SPSP client begins sending ILP packets to fulfill the payment.
+    1. The client will adjust their `sendMax` to reflect the amount they're willing to send.
+        * If present, `\|balance.current\|` SHOULD be used as the STREAM `sendMax`.
```

Comment:
I don't think this is correct. The client may be sending in a different currency.

The amount the client is willing to send is a business decision based on the expected amount `(balance.max - balance.current)`, the anticipated exchange rate, and the slippage the client will allow.

Reply:
You are right. That was blind copy-pasting. I'll fix it.